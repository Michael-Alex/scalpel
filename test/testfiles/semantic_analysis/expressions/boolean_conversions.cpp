bool b1 = 0;
const bool b2 = 1;
const bool& b3 = -2;
