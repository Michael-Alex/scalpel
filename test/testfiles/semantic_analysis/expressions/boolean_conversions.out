{
  "id": 0,
  "variables":
  [
    {
      "id": 0,
      "name": "b1",
      "type":
      {
        "category": "fundamental type",
        "type": "bool"
      },
      "default value":
      {
        "type": "conversion to bool",
        "source value":
        {
          "type": "int",
          "value": 0
        }
      }
    },
    {
      "id": 1,
      "name": "b2",
      "type":
      {
        "category": "cv-qualified type",
        "const": true,
        "qualified type":
        {
          "category": "fundamental type",
          "type": "bool"
        }
      },
      "default value":
      {
        "type": "conversion to bool",
        "source value":
        {
          "type": "int",
          "value": 1
        }
      }
    },
    {
      "id": 2,
      "name": "b3",
      "type":
      {
        "category": "reference",
        "qualified type":
        {
          "category": "cv-qualified type",
          "const": true,
          "qualified type":
          {
            "category": "fundamental type",
            "type": "bool"
          }
        }
      },
      "default value":
      {
        "type": "conversion to bool",
        "source value":
        {
          "type": "negation",
          "operand":
          {
            "type": "int",
            "value": 2
          }
        }
      }
    }
  ]
}
